Merge MetaMiner into the MockBukkit repository

Thorinwasher opened the pull request:
# Description
As time goes on with MockBukkit, more generated files have been necessary to produce. From the constant updating of the generator of these files `MetaMiner` I have realised that it becomes a hassle to keep the two projects synhronized. That's why I think it's a better idea to completely merge these projects into one repository.

This pull request does exactly that, MetaMiner has now been moved into the `extra/MetaMiner` folder in the MockBukkit repository. I also added a simple script that moves the generated files over to the mocbukkit project.

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Code follows existing style.
- [n/a] Unit tests added (if applicable).

Thorinwasher committed:
Initial commit

Thorinwasher committed:
Mines necessary keyed data

Thorinwasher committed:
Delete .idea directory

Thorinwasher committed:
Update .gitignore

Thorinwasher committed:
Implement fetching isSolid property

Thorinwasher committed:
Create PotionDataGenerator.java

Thorinwasher committed:
Implement enchantment keyed data

Thorinwasher committed:
Implement PotionEffectType keyed data

Thorinwasher committed:
Fix of invalid data generation

Thorinwasher committed:
Mine keyed object DamageType

Thorinwasher committed:
Generate tags

Thorinwasher committed:
Add new properties to keyed objects

Thorinwasher committed:
Move to gradle

svaningelgem committed:
Adding MaterialDataGenerator

svaningelgem committed:
Shutdown the server when finished. Easier for scripting the thing.

svaningelgem committed:
Merge remote-tracking branch 'origin/main' into adding-material-generation

Thorinwasher committed:
Merge pull request #1 from svaningelgem/adding-material-generation

Adding material_data.json generator

Thorinwasher committed:
Fix for plugin.yml and update to to 1.20.6

Thorinwasher committed:
Update scrapper accoridng to the 1.20.6 update of mockbukkit

Thorinwasher committed:
Sort generated files alphabetically

thelooter committed:
feat(1.21): update to 1.21 api

Signed-off-by: thelooter <evekolb2204@gmail.com>

Thorinwasher committed:
Merge pull request #3 from MockBukkit/update-1.21

feat(1.21): update to 1.21 api

Thorinwasher committed:
Add capability to scan methods for keyed instance generation

Thorinwasher committed:
Add byte method return type parsing ability

Thorinwasher committed:
Fix structure type being missed

thelooter committed:
Add more ItemType properties

thelooter committed:
Merge pull request #4 from MockBukkit/method-scanner

Add capability to scan methods for keyed instance data generation

Braekpo1nt committed:
added a list of items which satisfy the canEnchantItems() requirement to each enchantment's keyed data (#7)

Thorinwasher commented:
Just going to comment, that it's probably a good idea to merge the changes from metaminer before merging this into the main branch

Thorinwasher committed:
Translation mining (#8)

Thorinwasher committed:
Sort generated files alphabetically (#6)

4everTheOne committed:
Added README.md file to the project (#10)

* Added README.md file to the project

* Fixed icon in README.md

* fixup! Fixed icon in README.md

thelooter reviewed (changes requested):
Please convert it to a git submodule

4everTheOne committed:
Upgraded Paper version

Thorinwasher commented:
> Please convert it to a git submodule

I dissagree with that, with the stated reasons in my description, i.e. we need to keep these projects completely synched. Anyhow, I think this might be a somewhat heated topic, so we probably need to have a voice call about this.

thelooter committed:
Merge pull request #13 from MockBukkit/feat/more-itemtype

thelooter committed:
Merge pull request #11 from MockBukkit/fix/upgrade-metaminer-version

4everTheOne committed:
Added support for 1.21.3

Thorinwasher committed:
Add itemstack testdata (#14)

* fix(itemstack): Add itemstack testdata

* fix(itemstack): Test data generation with multiple interfaces

thelooter committed:
Merge pull request #15 from MockBukkit/fix/1.21.3-support

Added support for 1.21.3

4everTheOne committed:
Improved MethodDataScanner to support more types (#16)

Thorinwasher committed:
Add 'metaminer/' from commit 'f382ab7b2da4bf3c9adb282e4bc2a04426fa587b'

git-subtree-dir: metaminer
git-subtree-mainline: a6b20cc936259912602d438fece907d562ee7f8d
git-subtree-split: f382ab7b2da4bf3c9adb282e4bc2a04426fa587b

Thorinwasher committed:
fix(files): Remove unused files

Thorinwasher committed:
fix(git): Ignore metaminer files

Thorinwasher committed:
Fix add metaminer to build system

Thorinwasher committed:
fix(resource-gen): Fix wrong location of internal resources

Thorinwasher committed:
fix(readme): eula location

thelooter committed:
Merge branch 'v1.21' into metaminer
